en/docs/Learn/Tutorials/subscribe-to-an-api.md

```diff
@@ -1,195 +1,201 @@
+    After selecting a policy, Click __NEXT__ to continue.
 
-    ![]({{base_path}}/assets/attachments/103327837/103327819.png)
+5\. In the third step (__Generate Keys__),You can choose grant types which needs to be allowed for this application.You may select the grant types for __PRODUCTION__ and __SANDBOX__ environments.
+    ![](../../assets/img/Learn/application-subscription-key-gen-wizard-3rd-step.png)
 
-9.  Click **Generate Keys** to create an application access token. You can use this token to invoke all APIs that you subscribe to using the same application.
+!!! note
+    By default, the __Client Credentials__ grant type is used to generate the access token. Make sure the Client Credentials grant type is selected when generating keys from the UI. For more information on how to generate supported grant types of WSO2 API Manager, see [Token API](_Token_API_).
 
-        !!! tip
-    You can set a token validity period in the **Access token validity period** text box. By default, it is set to one hour. If you set a minus value (e.g., -1), the token never expires.
+!!! note
+    If you have a supported callback URL that sends a callback to a specific server or program soon after your application request is sent, you can specify it under the **Callback URL** field.
+    Click __NEXT__ after selecting the required grant types.
+    Application key and secret will be generated, When you click next.
 
-        !!! info
-    By default, the Client Credentials grant type is used to generate the access token. Make sure the Client Credentials grant type is selected when generating keys from the UI. For more information on how to generate supported grant types of WSO2 API Manager, see [Token API](_Token_API_) .
+6\.  In this step, You can provide the access token validity period and set of scopes.By default, validity period is set to one hour. If you set a minus value (e.g., -1), the token never expires.
+    ![](../../assets/img/Learn/application-subscription-key-gen-wizard-4th-step.png)
 
-        !!! tip
+!!! tip
     Access Tokens with specific Scopes
 
     Access tokens can be generated for specific scopes. A scope acts as a limiting factor on what API resources can be accessed using a token. To generate an access token corresponding to a scope, use the drop down menu under **Scopes** and select the required scope parameter.
 
-        !!! note
-    If you are using WSO2 Identity Server 5.6.0 as the Key Manager for your API Manager deployment, generating keys will result in the creation of a [Service Provider](https://docs.wso2.com/display/IS560/Configuring+a+Service+Provider) in the Identity Server.
+!!! note
+    If you are using WSO2 Identity Server 5.9.0 as the Key Manager for your API Manager deployment, Clicking __NEXT__ will result in the creation of a [Service Provider](https://docs.wso2.com/display/IS590/Configuring+a+Service+Provider) in the Identity Server.
```

Comment:
Is this IS URL correct? However, there is a https://is.docs.wso2.com/en/5.9.0/setup/configuring-a-sp-and-idp-using-configuration-files/

Reply:
Thanks for pointing out this, I think the correct URL is

https://is.docs.wso2.com/en/5.9.0/learn/adding-and-configuring-a-service-provider/

Reply:
fixed in https://github.com/wso2/docs-apim/pull/326/commits/d3856d812d2db0cbb9b24c6374887d904f23ba61